tox.ini

```diff
@@ -60,5 +62,34 @@ deps =
     -r{toxinidir}/requirements.txt
     pyfakefs
 commands =
-	coverage run --source={[vars]lib_path} -m pytest -v --tb native {posargs}
+	coverage run --source={[vars]lib_path} \
+        -m pytest --ignore={[vars]tst_path}integration -v --tb native {posargs}
     coverage report
+
+[testenv:integration]
+description = Build a LXD container for integration tests then run the tests
+allowlist_externals = 
+    lxc
+    bash
+commands =
+    ; Create a LXC container with the relevant packages installed
+    bash -c 'lxc launch -qe ubuntu:focal {[vars]lxd_name} --config=user.user-data="$(<{toxinidir}/tests/integration/test-instance.yaml)"'
+    ; Wait for the cloud-init process to finish
+    lxc exec {[vars]lxd_name} -- bash -c "cloud-init status -w >/dev/null 2>&1"
+    ; Copy all the files needed for integration testing
+    lxc file push -qp {toxinidir}/tox.ini {[vars]lxd_name}/root/operator-libs-linux/
+    lxc file push -qp {toxinidir}/pyproject.toml {[vars]lxd_name}/root/operator-libs-linux/
+    lxc file push -qpr {toxinidir}/tests {[vars]lxd_name}/root/operator-libs-linux/
+    lxc file push -qpr {toxinidir}/lib {[vars]lxd_name}/root/operator-libs-linux/
+    ; Run the tests
+    lxc exec {[vars]lxd_name} -- tox -c /root/operator-libs-linux/tox.ini -e integration-tests
+commands_post =
+    -lxc stop {[vars]lxd_name}
```

Comment:
I like this approach a lot. We can be flexible about where and when we run the tests, without having to actually change any of the tests themself. I suppose that we are relying on a usable version of Python being in the testing container. But again, I don't think that's a super big concern. And it's not like we're using a bunch of Python version specific features.

Reply:
The usable version of Python should get dragged in automatically by cloud-init when it installs `python3-pip` :)

Reply:
I also like this approach. I was a bit worried originally, but the fact that we give you a clean place to do real mutating changes is a nice way to validate everything works as intended.